src/realm/query.cpp

```diff
@@ -1677,7 +1677,7 @@ size_t Query::count(const DescriptorOrdering& descriptor)
 }
 
 // todo, not sure if start, end and limit could be useful for delete.
-size_t Query::remove()
+size_t Query::remove() const
```

Comment:
I could see an argument for not making this `const`. I chose to mark it as such because while it does modify the underlying table, it doesn't change the `Query` itself.